suite-native/module-device-settings/src/navigation/DevicePinProtectionStackNavigator.tsx

```diff
@@ -1,25 +1,60 @@
+import { useSelector } from 'react-redux';
+
 import { createNativeStackNavigator } from '@react-navigation/native-stack';
 
+import { selectDeviceButtonRequestsCodes } from '@suite-common/wallet-core';
 import {
     DevicePinProtectionStackParamList,
     DevicePinProtectionStackRoutes,
     stackNavigationOptionsConfig,
 } from '@suite-native/navigation';
 
 import { ContinueOnTrezorScreen } from '../screens/ContinueOnTrezorScreen';
+import { EnterPinScreen } from '../screens/EnterPinScreen';
 
 export const DevicePinProtectionStack =
     createNativeStackNavigator<DevicePinProtectionStackParamList>();
 
-export const DevicePinProtectionStackNavigator = () => (
-    <DevicePinProtectionStack.Navigator
-        initialRouteName={DevicePinProtectionStackRoutes.ContinueOnTrezor}
-        screenOptions={stackNavigationOptionsConfig}
-    >
-        <DevicePinProtectionStack.Screen
-            name={DevicePinProtectionStackRoutes.ContinueOnTrezor}
-            component={ContinueOnTrezorScreen}
-        />
-        {/*TODO: To be extended for T1*/}
-    </DevicePinProtectionStack.Navigator>
-);
+export const DevicePinProtectionStackNavigator = () => {
+    const buttonRequestCodes = useSelector(selectDeviceButtonRequestsCodes);
+    const lastButtonRequestCode = buttonRequestCodes[buttonRequestCodes.length - 1];
+
+    const isEnterCurrentPin = lastButtonRequestCode === 'PinMatrixRequestType_Current';
+    const isEnterNewPin = lastButtonRequestCode === 'PinMatrixRequestType_NewFirst';
+    const isConfirmNewPin = lastButtonRequestCode === 'PinMatrixRequestType_NewSecond';
+
+    return (
+        <DevicePinProtectionStack.Navigator
+            initialRouteName={DevicePinProtectionStackRoutes.ContinueOnTrezor}
+            screenOptions={stackNavigationOptionsConfig}
+        >
+            {!isEnterCurrentPin && !isEnterNewPin && !isConfirmNewPin && (
```

Comment:
nit: can you, maybe, make this into one const that explains by its name why this condition or add comment to explain?